app/services/samples/batch_file_import_service.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'roo'
+
+module Samples
+  # Service used to batch create samples via a file
+  class BatchFileImportService < BaseSpreadsheetImportService
+    def initialize(namespace, user = nil, blob_id = nil, params = {})
+      @sample_name_column = params[:sample_name_column]
+      @project_puid_column = params[:project_puid_column]
+      @sample_description_column = params[:sample_description_column]
+      required_headers = [@sample_name_column, @project_puid_column]
+      super(namespace, user, blob_id, required_headers, 0, params)
+    end
+
+    def execute
+      authorize! @namespace, to: :update_sample_metadata?
+      validate_file
+      perform_file_import
+    rescue FileImportError => e
+      @namespace.errors.add(:base, e.message)
+      {}
+    end
+
+    protected
+
+    def perform_file_import # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
+      response = {}
+      parse_settings = @headers.zip(@headers).to_h
+
+      @spreadsheet.each_with_index(parse_settings) do |data, index|
+        next unless index.positive?
+
+        # TODO: handle metadata
```

Comment:
I'm thinking this can be done in the process_sample_row once the sample is persisted